clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -462,12 +462,10 @@ public void prepareResponse(RequestMatcher matcher, AbstractResponse response, b
     }
 
     /**
-     * Raise an unsupported version error on the next request if it matches the given matcher.
-     * If the matcher does not match, {@link KafkaClient#send(ClientRequest, long)} will throw IllegalStateException.
-     * @param matcher The request matcher to apply
+     * Raise an unsupported version error on the next request.
      */
-    public void prepareUnsupportedVersionResponse(RequestMatcher matcher) {
```

Comment:
That allowed us to raise unsupported version for specific request. Maybe we should make it work rather then remove it.